### PR TITLE
feat: [PL-24533]: Add new interface in gitSyncGRPC Client to getFileContent() from SCM

### PR DESCRIPTION
**Pull request by mohitg0795**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)

### CODE REVIEW

File: 953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto [hunk 1]
```diff
@@ -85,3 +86,34 @@ message BranchDetails {
...
+
+message GitMetaData {
+  string repoName = 1;
+  string branchName = 2;
+  string commitId = 3;
+  string filePath = 4;
+  string blobId = 5;
+}
+
+message ScopeIdentifiers {
```
Comment (abhinav-harness) [inline 953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto:115]: move it to commons or i think it already has it.

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/github/GithubConnectorDTO.java [hunk 2]
```diff
@@ -93,4 +93,13 @@ public List<DecryptableEntity> getDecryptableEntities() {
...
     return ConnectorType.GITHUB;
   }
+
+  @Override
+  public String getGitConnectionUrl(String repoName) {
+    if (connectionType == GitConnectionType.REPO) {
+      // check if we need to put any validation for repoName in this case
+      return getUrl();
+    }
+    return getUrl() + "/" + repoName;
```
Comment (abhinav-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/github/GithubConnectorDTO.java:103]: url already ending with / while saving?

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
changes

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/bitbucket/BitbucketConnectorDTO.java [hunk 3]
```diff
@@ -91,6 +91,9 @@ public ConnectorType getConnectorType() {
 
   @Override
   public String getGitConnectionUrl(String repoName) {
-    return "";
+    if (connectionType == GitConnectionType.REPO) {
+      return getUrl();
+    }
+    return getUrl() + "/" + repoName;
```
Comment (namanvermah) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/bitbucket/BitbucketConnectorDTO.java:97]: What if the URL in getUrl() already has a slash at the end? Also, should we end the full repo with a slash?

**Comment by mohitg0795** [comment]
*[953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto, line 115 at r2](https://reviewable.io/reviews/harness/harness-core/32248#-N-tHOf2RSQQwPu0ei9--r2-115:-N-vzUnL2njBYoRVqJ8F:b-84847m) ([raw file](https://github.com/harness/harness-core/blob/660f42bc431bd342b61b54b5039c3623f0c3fd96/953-git-sync-commons/src/main/proto/io/harness/gitsync/harness_to_git.proto#L115)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

move it to commons or i think it already has it.
</blockquote></details>

Moved

**Comment by mohitg0795** [comment]
*[954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/bitbucket/BitbucketConnectorDTO.java, line 97 at r3](https://reviewable.io/reviews/harness/harness-core/32248#-N-tHOf2RSQQwPu0ei93-r3-97:-N-vzWQ9DpXlxrFEQz9w:b-a4j25l) ([raw file](https://github.com/harness/harness-core/blob/af67e568c11a9eb777a4ffdd26932f2c3dfbf68f/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/bitbucket/BitbucketConnectorDTO.java#L97)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

What if the URL in getUrl() already has a slash at the end? Also, should we end the full repo with a slash?
</blockquote></details>

I think double slashes don't create any issue, but anyways better to handle it. Added handling.

**Comment by mohitg0795** [comment]
*[954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/github/GithubConnectorDTO.java, line 103 at r2](https://reviewable.io/reviews/harness/harness-core/32248#-N-tHOf2RSQQwPu0ei95-r2-103:-N-vzcbq791rELW03Gkd:b-a4j25l) ([raw file](https://github.com/harness/harness-core/blob/660f42bc431bd342b61b54b5039c3623f0c3fd96/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/github/GithubConnectorDTO.java#L103)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

url already ending with / while saving?
</blockquote></details>

I think double slashes don't create any issue, but anyways better to handle it. Added handling.

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java, line 141 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w63Zl-dwz_boRiK6A:-N-w63Zl-dwz_boRiK6B:bq2cmj0) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java#L141)):*
> ```Java
>       String accountIdentifier, String orgIdentifier, String projectIdentifier, String connectorRef) {
>     ScmConnector gitConnectorConfig =
>         getScmConnector(accountIdentifier, orgIdentifier, projectIdentifier, connectorRef);
> ```

There is one more function which takes repo and branch also iin consideration. We should be using that getSCM call

If the main git connector is not stored in default branch and it is in different repo and branch, then we need to pass repo and branch

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java, line 182 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w7M8f7dx3cfnObk9Q:-N-w7M8f7dx3cfnObk9R:b-rigs38) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java#L182)):*
> ```Java
>   }
> 
>   public ScmConnector getScmConnector(
> ```

You have copied this function from somewhere ?

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/HarnessToGitHelperServiceImpl.java, line 327 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w7tW28_9HMmSwuhBL:-N-w7tW28_9HMmSwuhBM:b-4y0vmc) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/HarnessToGitHelperServiceImpl.java#L327)):*
> ```Java
> 
>   @Override
>   public GetFileResponse getFile(GetFileRequest getFileRequest) {
> ```

getFile is a new function, is it not required by other scm providers ?

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java, line 141 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w63Zl-dwz_boRiK6A:-N-w8m3V0AiTuTrhtuQC:bz1wdad) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java#L141)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

There is one more function which takes repo and branch also iin consideration. We should be using that getSCM call

If the main git connector is not stored in default branch and it is in different repo and branch, then we need to pass repo and branch
</blockquote></details>

This PR is in reference to Git Simplification feature, not our git-sync. So, there's no repo and branch for a connector here.

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java, line 182 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w7M8f7dx3cfnObk9Q:-N-w8va893EgGBPIGvW8:bw1duhy) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitSyncConnectorHelper.java#L182)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

You have copied this function from somewhere ?
</blockquote></details>

Not exactly copied, but did similar to what had been done before.

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/HarnessToGitHelperServiceImpl.java, line 327 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w7tW28_9HMmSwuhBL:-N-w92-D1nHc4SaiPZZl:b-luqsl6) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/HarnessToGitHelperServiceImpl.java#L327)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

getFile is a new function, is it not required by other scm providers ?
</blockquote></details>

This is in harnessToGitGRPC service, so its agnostic of scm providers.

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java, line 184 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w9YtN1BtuZSPrSbyY:-N-w9YtN1BtuZSPrSbyZ:bp5cde8) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java#L184)):*
> ```Java
>     try {
>       return FileContent.parseFrom(gitFileTaskResponseData.getFileContent());
>     } catch (InvalidProtocolBufferException e) {
> ```

also log the error message

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 235 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w9kwV2DFFmJBGKSoY:-N-w9kwV2DFFmJBGKSoZ:b-nlyrwm) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L235)):*
> ```Java
>   @GET
>   @Path("get-file")
>   @ApiOperation(value = "get file", nickname = "getFile")
> ```

G should be capital in get file ?

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 238 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w9rt60e-XK4W8u_0c:-N-w9rt60e-XK4W8u_0d:b-in5aq7) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L238)):*
> ```Java
>   @Hidden
>   @Operation(operationId = "getFile", summary = "get file",
>       responses = { @io.swagger.v3.oas.annotations.responses.ApiResponse(description = "Successfully created a PR") },
> ```

Description is not related

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 247 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wA-N07KOLD1RW0RSa:-N-wA-N14Fzg7eiJCRLj:b7x6q5a) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L247)):*
> ```Java
>       @Parameter(description = PROJECT_PARAM_MESSAGE) @QueryParam(
>           NGCommonEntityConstants.PROJECT_KEY) @ProjectIdentifier String projectIdentifier,
>       @Parameter(description = "Repo Name") @QueryParam("RepoName") @NotBlank @NotNull String repoName,
> ```

The query param should not start with a capital letter

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 252 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wA6PO-s_1mfbF_c-q:-N-wA6PO-s_1mfbF_c-r:b-2guby3) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L252)):*
> ```Java
>       @Parameter(description = "File Path") @QueryParam(YamlConstants.FILE_PATH) @NotBlank @NotNull String filePath,
>       @Parameter(description = "Commit Id") @QueryParam(YamlConstants.COMMIT_ID) String commitId,
>       @Parameter(description = "Connector Ref") @QueryParam("ConnectorRef") String connectorRef) {
> ```

same here

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 35 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[970-ng-commons/src/main/proto/io/harness/scope_identifiers.proto, line 11 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wB7GZC4tI7FF5pT6C:-N-wB7GZC4tI7FF5pT6D:bg78wz7) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/970-ng-commons/src/main/proto/io/harness/scope_identifiers.proto#L11)):*
> ```protobuf
> option go_package = "970_ng_commons";
> 
> message ScopeIdentifiers {
> ```

A proto class might already be present for this use case. Can we use that file
___
*[980-commons/src/main/java/io/harness/utils/FilePathUtils.java, line 59 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wAqn9CiyJt70c5UFO:-N-wAqn9CiyJt70c5UFP:b-4huc8s) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/980-commons/src/main/java/io/harness/utils/FilePathUtils.java#L59)):*
> ```Java
> 
>   public static String addEndingSlashIfMissing(String filePath) {
>     if (isNotEmpty(filePath) && filePath.charAt(filePath.length() - 1) == '/') {
> ```

We can use endsWith here

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 235 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w9kwV2DFFmJBGKSoY:-N-wGhP288ryL4qNJPUE:bruq508) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L235)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

G should be capital in get file ?
</blockquote></details>

This API isn't exposed anywhere, its redundant swagger details. This API will be removed, added only for testing as mentioned in comments.

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 238 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w9rt60e-XK4W8u_0c:-N-wHYIXC4mcDPw0PFPo:bruq508) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L238)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Description is not related
</blockquote></details>

This API isn't exposed anywhere, its redundant swagger details. This API will be removed, added only for testing as mentioned in comments.

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 247 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wA-N07KOLD1RW0RSa:-N-wHhbG58hN60uSoL5V:bruq508) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L247)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

The query param should not start with a capital letter
</blockquote></details>

This API isn't exposed anywhere, its redundant swagger details. This API will be removed, added only for testing as mentioned in comments.

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java, line 252 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wA6PO-s_1mfbF_c-q:-N-wHiueEcgJYpns04Ze:bruq508) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/remote/ScmFacilitatorResource.java#L252)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

same here
</blockquote></details>

This API isn't exposed anywhere, its redundant swagger details. This API will be removed, added only for testing as mentioned in comments.

**Comment by mohitg0795** [comment]
*[970-ng-commons/src/main/proto/io/harness/scope_identifiers.proto, line 11 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wB7GZC4tI7FF5pT6C:-N-wHmJt1PPyy6P4xIoU:b853dms) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/970-ng-commons/src/main/proto/io/harness/scope_identifiers.proto#L11)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

A proto class might already be present for this use case. Can we use that file
</blockquote></details>

No proto class exists for this use case.

**Comment by mohitg0795** [comment]
*[980-commons/src/main/java/io/harness/utils/FilePathUtils.java, line 59 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-wAqn9CiyJt70c5UFO:-N-wJ2MW65W_B2hIs-9e:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/980-commons/src/main/java/io/harness/utils/FilePathUtils.java#L59)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

We can use endsWith here 
</blockquote></details>

done

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java, line 184 at r4](https://reviewable.io/reviews/harness/harness-core/32248#-N-w9YtN1BtuZSPrSbyY:-N-wJSTnEAen5uytczzS:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/70b737061dff5da4d907822260551bd6d8291b24/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java#L184)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

also log the error message
</blockquote></details>

done

**Review by DeepakPatankar — COMMENTED** [review]
> Reviewed 12 of 28 files at r2, 11 of 11 files at r3, 5 of 8 files at r4, 10 of 10 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 38 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 38 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248#-:-N00BQ6BB2G86V868Vex:bnfp4nl)* status: 38 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 38 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java, line 172 at r6](https://reviewable.io/reviews/harness/harness-core/32248#-N00WwQf6Y1ijMCDe5cF:-N00WwQf6Y1ijMCDe5cG:bi4spit) ([raw file](https://github.com/harness/harness-core/blob/78b5268047c2c49c79a3cfd1ac81b0d744745c19/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java#L172)):*
> ```Java
>       String connectorRef, String repoName, String branchName, String filePath, String commitId) {
>     ScmConnector connector =
>         getScmConnector(accountIdentifier, orgIdentifier, projectIdentifier, connectorRef, repoName);
> ```

This is the decrypted connector but we do not send decrypted connector in delegate task. @deepakpatankar do you remember the reason of not sending decrypted connector here.

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 12 of 28 files at r2, 1 of 11 files at r3, 5 of 8 files at r4, 9 of 10 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248#-:-N00ZKlYA-DruHdi5qOP:bnfp4nl)* status: 11 file(s) to review. 2 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java, line 172 at r6](https://reviewable.io/reviews/harness/harness-core/32248#-N00WwQf6Y1ijMCDe5cF:-N00ZlJx8_mJw4u8Bc0G:bul4xyq) ([raw file](https://github.com/harness/harness-core/blob/78b5268047c2c49c79a3cfd1ac81b0d744745c19/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java#L172)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

This is the decrypted connector but we do not send decrypted connector in delegate task. @deepakpatankar do you remember the reason of not sending decrypted connector here.
</blockquote></details>

@bhavya181 can you resolve this comment. It blocks my PR for now. As discussed, we will check into it and then make changes if required. @deepakpatankar can help to answer our query. 
Btw Deepak, I tested the code and it worked fine. We are curious why we send usual connector and how should we go about in the future.

Just request to make the thread as "discussion" instead of 'blocking". Thanks.

**Review by bhavya181 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 12 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java, line 172 at r6](https://reviewable.io/reviews/harness/harness-core/32248#-N00WwQf6Y1ijMCDe5cF:-N00_Ns63fJyfvgoy0Rt:bppahvz) ([raw file](https://github.com/harness/harness-core/blob/78b5268047c2c49c79a3cfd1ac81b0d744745c19/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java#L172)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

@bhavya181 can you resolve this comment. It blocks my PR for now. As discussed, we will check into it and then make changes if required. @deepakpatankar can help to answer our query. 
Btw Deepak, I tested the code and it worked fine. We are curious why we send usual connector and how should we go about in the future.

Just request to make the thread as "discussion" instead of 'blocking". Thanks.
</blockquote></details>

sure, I am resolving it for now, we will discuss it internally.

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 12 of 28 files at r2, 11 of 11 files at r3, 5 of 8 files at r4, 8 of 10 files at r5, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 2 file(s) to review. 1. unresolved discussions.

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 12 of 28 files at r2, 1 of 11 files at r3, 5 of 8 files at r4, 8 of 10 files at r5, 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 3 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by vardanbansal-harness — DISMISSED** [review]
> Reviewed 12 of 28 files at r2, 1 of 11 files at r3, 5 of 8 files at r4, 8 of 10 files at r5, 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 1. unresolved discussions.

**Comment by mohitg0795** [comment]
trigger ti0

**Comment by mohitg0795** [comment]
trigger ti1

**Comment by mohitg0795** [comment]
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java, line 172 at r6](https://reviewable.io/reviews/harness/harness-core/32248#-N00WwQf6Y1ijMCDe5cF:-N04PseY6zAT2WpJJ6qa:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/78b5268047c2c49c79a3cfd1ac81b0d744745c19/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmDelegateFacilitatorServiceImpl.java#L172)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

sure, I am resolving it for now, we will discuss it internally.
</blockquote></details>

Done.

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 12 of 28 files at r2, 1 of 11 files at r3, 5 of 8 files at r4, 8 of 10 files at r5, 1 of 3 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 12 of 28 files at r2, 1 of 11 files at r3, 5 of 8 files at r4, 8 of 10 files at r5, 1 of 3 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248#-:-N04_1hL9SehpPcZlxpg:bnfp4nl)* status: 1. unresolved discussions.

**Review by bhavya181 — APPROVED** [review]
> Reviewed 12 of 28 files at r2, 11 of 11 files at r3, 5 of 8 files at r4, 8 of 10 files at r5, 1 of 3 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: 1. unresolved discussions.

**Review by bhavya181 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32248)* status: :shipit: complete! ✓

**Pull request merged**